config/cht/tasks.js

```diff
@@ -0,0 +1,528 @@
+const extras = require('./nools-extras');
+
+const {
+  today,
+  getNewestPregnancyTimestamp, 
+  getNewestDeliveryTimestamp,
+  isAlive,
+  isFormArraySubmittedInWindow,
+  getDateMS,
+  getDateISOLocal,
+  getTimeForMidnight,
+  getMostRecentLMPDateForPregnancy,
+  addDays,
+  getRecentANCVisitWithEvent,
+  isPregnancyTaskMuted,
+} = extras;
+
+module.exports = [
+
+  //ANC
+  //ANC Home Visit: 12, 20, 26, 30, 34, 36, 38, 40 weeks (Known LMP)
+  {
+    icon: "icon-pregnancy",
+    title: "task.anc.pregnancy_home_visit.title",
+    appliesTo: "reports",
+    appliesToType: ["pregnancy"],
+    appliesIf: function (c, r) {// If LMP date is known
+      const recentLMP = getMostRecentLMPDateForPregnancy(c, r);
+      let x = recentLMP !== null;
+      //console.log('l', new Date(recentLMP));
+      //console.log('applies', x, c.contact.name);
+      return x;
+    },
+
+    resolvedIf: function (c, r, event, dueDate) {
+      //console.log('ddr', dueDate);
+      if (r.reported_date < getNewestDeliveryTimestamp(c)) return true;//delivery form submitted
+      if (r.reported_date < getNewestPregnancyTimestamp(c)) return true;//old pregnancy report
+
+      //miscarriage or abortion
+      if (getRecentANCVisitWithEvent(c, r, 'abortion') || getRecentANCVisitWithEvent(c, r, 'miscarriage')) return true;
+
+      //Due date older than reported day
+      if(dueDate <= getTimeForMidnight(r.reported_date)) return true;
+
+      //clear tasks
+      if (isPregnancyTaskMuted(c)) return true;
+
+      let x = isFormArraySubmittedInWindow(c.reports, ['pregnancy_home_visit'],
+        Math.max(addDays(dueDate, -event.start).getTime(), r.reported_date),
+        addDays(dueDate, event.end + 1).getTime());
+      //console.log('resolved by visit report', x);
+      //let x = false;
+      return x;
+    },
+
+    actions: [
+      {
+        type: "report",
+        form: "pregnancy_home_visit",
+        label: "Pregnancy home visit",
+        modifyContent: function (content, c, r) {
+          content.source = 'task';
+          content.source_id = r._id;
+          content.contact = c.contact;
+          content.pregnancy_uuid = r._id;
+          //content.lmp_date_8601 = r.fields.lmp_date_8601;
+          //content.edd_8601 = r.fields.edd_8601;
+          //content.weeks_since_lmp = r.fields.weeks_since_lmp;
+        }
+      }
+    ],
+    events: [
+      {
```

Comment:
You can avoid a lot of redundant boilerplate here by creating a function which generates one of these event objects, so for example:

```
const generateEvent = day => ({
        id: "pregnancy-home-visit",
        start: 7,
        end: 14,
        dueDate: function (e, c, r) {
          let x = addDays(getMostRecentLMPDateForPregnancy(c, r), day * 7);
          return x;
        }
})
```

Then you can have:

```
events: [
  generateEvent(12),
  generateEvent(20),
  generateEvent(26),
  generateEvent(30),
  ...etc...
]
```

Or even easier:

```
events: [12, 20, 26, 30, ....].map(generateEvent)
```